packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -29,7 +29,16 @@ export class FabricBase extends React.Component<
     });
     const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
 
-    return <div {...divProps} className={classNames.root} ref={this._rootElement} />;
+    let fabricComponent;
+    // apply theme to just the div only if applyTheme is true, otherwise do not apply theme at all
+    fabricComponent = (
+      <div {...divProps} className={this.props.applyTheme ? classNames.rootThemed : classNames.root} ref={this._rootElement} />
+    );
+    if (this.props.applyThemeToBody) {
+      // apply theme to the body of the whole page
+      document.body.classList.add('ms-Fabric-body-themed');
```

Comment:
Multiple problems here.

1. You should never access `document` or `window` until the component is mounted. Otherwise you break SSR.

2. `document.body` is the host frame, every time. It seems like on mount, you should call `getDocument` with the _`rootElement` prop to get the actual document  the content is hosted inside of.

3. The classname used should be generated from the styling file, not a global classname.

4. Once you've added the classname, you're never cleaning it up. This is tricky. You probably need a refcount solution to have the last add remove the classname.

On unmount this is still applied. Hmmm. Also, shouldn't there be

Reply:
I don't think this will work, because you are adding the global name and not the generated one. Please test that applying to the body actually applies the generated classname to the body.

Reply:
Changed it so that now it's adding the generated classname `bodyThemed` - trying to add a snapshot test to confirm this (I see that it does it successfully when I wrap one of the component examples with this and inspect the dom but want to of course add a test too) however I get a jest error saying `The token provided contains HTML space characters, which are not valid in tokens.` which I think might be due to the removing on unmount - looking into this.